### DISCUSSION TITLE
cron does not start inside a docker container with alma linux, no failures or errors to be seen

### DISCUSSION THREAD

**Discussion by sshakhamuri-godaddy** [community/community#137971] [Programming Help]
### Select Topic Area

General

### Body

I see that the command to start cron runs successfully and all the necessary packages are installed, however when my container runs I see that cron isn't running. What am I missing?

This is my docker script 
`RUN yum install -y epel-release && \
    yum install -y crontabs cronie cronie-anacron

COPY cleanup /etc/cron.d/cleanup
RUN chmod 0644 /etc/cron.d/cleanup
RUN chown root:root /etc/cron.d/cleanup

RUN mkdir -p /var/log/cron && \
    chown -R root:root /var/log/cron

RUN /usr/sbin/crond`

**Comment by AVMiraga** [comment]
```
RUN yum install -y epel-release &&
yum install -y crontabs cronie cronie-anacron

COPY cleanup /etc/cron.d/cleanup
RUN chmod 0644 /etc/cron.d/cleanup
RUN chown root:root /etc/cron.d/cleanup

RUN mkdir -p /var/log/cron &&
chown -R root:root /var/log/cron

CMD /usr/sbin/crond
```

> [CMD](https://docs.docker.com/reference/dockerfile/#cmd) command param1 param2 (shell form)

**Comment by Thxmxs** [comment]
It looks like you're trying to install and configure cron in a Docker container. However, there's an issue with how you're starting crond. In Docker, processes started in RUN commands (such as RUN /usr/sbin/crond) will not continue running when the container is started because RUN commands are executed during the build phase, not when the container is running.

To fix this, you need to ensure that crond starts when the container starts. You can do this by specifying the CMD or ENTRYPOINT in your Dockerfile.

Here’s how you can adjust your Dockerfile:

Modify the Dockerfile to ensure that crond starts in the running container: Replace the line RUN /usr/sbin/crond with a proper CMD or ENTRYPOINT command to start crond when the container runs.
Dockerfile

RUN yum install -y epel-release && \
    yum install -y crontabs cronie cronie-anacron

COPY cleanup /etc/cron.d/cleanup
RUN chmod 0644 /etc/cron.d/cleanup && \
chown root:root /etc/cron.d/cleanup
RUN mkdir -p /var/log/cron && \
chown -R root:root /var/log/cron
RUN crontab /etc/cron.d/cleanup
CMD ["/usr/sbin/crond", "-n"]
